lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
The reason is that in #7051, people seemed to agree that throwing a `RangeError` on things like `Buffer(-1)` & co. would be a good idea, especially as it leads to consistency in the behaviours of `Buffer()`, `Buffer.alloc()`, `Buffer.allocUnsafe()` and `Buffer.allocUnsafeSlow()`.

You can feel free to disagree and keep it in this PR as-is, too, but I’d expect that to come up in the discussion on #7079.